docker/frontend.prod.dockerfile

```diff
@@ -1,9 +1,20 @@
-FROM node:latest
+FROM node:12.18.1
 
-WORKDIR /app
+RUN npm install -g serve
 
-COPY . /app
+WORKDIR /temp
 
-RUN npm install -g serve
+COPY package*.json ./
+
+RUN npm install
```

Comment:
Could we actually make everything up to this line it's own dockerfile that only gets rebuilt if the package json files change, and then use that container as a base for the build/serve container?

Not sure if it's really necessary though so would love to hear your input on it

Reply:
Docker uses a caching system on every command, the commands were done this way to actually take advantage of this already. See this page for more info https://docs.docker.com/develop/develop-images/dockerfile_best-practices/.

Reply:
I am aware of the cache system.

This is true for local development but to my knowledge the Github Actions runners do not maintain the docker cache, or if so they do not do so for very long. This would be a way of utilizing a pseudo-cache system to reduce the build times for our Github Actions builds.

Reply:
I would argue the following:
a) This is extra complexity and unnecessary since github actions build time is not that long nor a constraint we need to worry about.
b) Github actions will likely implement a docker layer cache sometime in the future.
c) This is out of scope of this PR either way and should be done in a subsequent issue if needed.

Reply:
Added #35 since you find it to be out of scope for this PR